tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -660,7 +660,7 @@ void cbmc_populate_s2n_handshake_parameters(struct s2n_handshake_parameters *s2n
     CBMC_ENSURE_REF(s2n_handshake_parameters);
     cbmc_populate_s2n_pkey(&(s2n_handshake_parameters->server_public_key));
     cbmc_populate_s2n_pkey(&(s2n_handshake_parameters->client_public_key));
-    cbmc_populate_s2n_signature_scheme(&(s2n_handshake_parameters->conn_sig_scheme));
+    cbmc_populate_s2n_signature_scheme(&(s2n_handshake_parameters->server_cert_sig_scheme));
```

Comment:
Nice catch-- I missed that replacement. I've fixed it to allocate instead of just populate.

CBMC is much less strict in its compilation than our main builds, and doesn't use "-Werror" ([Makefile](https://github.com/aws/s2n-tls/blob/2d053022fa7051424ec545325fcdc6c4bd47480b/tests/cbmc/proofs/Makefile.common#L302)). "Incompatible pointer types" is only a warning by default.

I pushed to a branch on my fork with a CBMC makefile [that added "-Werror"](https://github.com/lrstewart/s2n/commit/8faadb4c559163dd71d516212f0e2437a1ee8b96). CBMC failed for almost all proofs with the error you'd expect: 
```
/home/runner/work/s2n-tls/s2n-tls/tests/cbmc/sources/make_common_datastructures.c:663:1: warning: warning: conversion from 'const struct s2n_signature_scheme **' to 'struct s2n_signature_scheme *': incompatible pointer types
cbmc_populate_s2n_signature_scheme(&(s2n_handshake_parameters->server_cert_sig_scheme));
```

I then created another branch with my fixed version and the same [updated Makefile](https://github.com/aws/s2n-tls/pull/4244/commits/f42b9eeaea6dd17c5a8d8c45b31735c7c8f1e6d3). CBMC did still [fail for one proof](https://github.com/aws/s2n-tls/actions/runs/6491747806?pr=4244), but for an unrelated reason (some X509 related conversions-- you can download the results for the full error). I'll open an issue to address that problem and probably turn on "-Werror" so we don't miss more problems.

Reply:
https://github.com/aws/s2n-tls/issues/4246

Reply:
Wow, I'm surprised that's only a warning by default!